tests/common/include/tests/common/tools/create_toy_geometry.hpp

```diff
@@ -225,13 +225,13 @@ inline void create_barrel_modules(context_t &ctx, volume_type &vol,
     // loop over the z bins
     for (size_t z_bin = 0; z_bin < size_t(n_z_bins); ++z_bin) {
         // prepare z and r
-        double m_z = z_start + z_bin * z_step;
-        double m_r = (z_bin % 2) != 0u
+        scalar m_z = z_start + z_bin * z_step;
+        scalar m_r = (z_bin % 2) != 0u
                          ? cfg.layer_r - 0.5 * cfg.m_radial_stagger
                          : cfg.layer_r + 0.5 * cfg.m_radial_stagger;
         for (size_t phiBin = 0; phiBin < size_t(n_phi_bins); ++phiBin) {
             // calculate the current phi value
-            double m_phi = min_phi + phiBin * phi_step;
+            scalar m_phi = min_phi + phiBin * phi_step;
             m_centers.push_back(
                 point3{static_cast<scalar>(m_r * std::cos(m_phi)),
```

Comment:
I am not sure all of these ```static_casts``` are strictly necessary anymore (might be for the one with ```std::cos``` etc in them)

Reply:
Right. I removed static_casts for some lines now